### DISCUSSION TITLE
how often can i run this script in a Minute

### DISCUSSION THREAD

**Discussion by justnils** [svenerbe/ecoflow_dynamic_power_adjustment#12] [Q&A] [answered]
Hi,
i just installed your script. Seems to work great.
Question i have:
How many time can i run this script in a minute?
Do you know the limitation of ecoflow calling the api?

Thanks
-Nils-

**Comment by svenerbe** [comment, marked as answer]
i run the script every time when i get new value from my grid sensor (update every 5 sec). So far i don't see any limitation from EcoFlow.

**Reply by bragg2012** [reply]
how you finding the 5 sec?  I've just changed mine to 10 sec as I found it was going a bit all over the place on 5 sec and wouldn't settle at 0.  My meter also updates/refreshes every 5 sec so not sure what was going on there.

**Reply by DauntlessEagle** [reply]
I find no matter what set 2 seconds 5 seconds 10 / 20 the script does a least 1 minute to feed back to the Powerstream. My smart meter updates also every 5 seconds via mqtt as it pinpoint accurate.

Anything in the python code we need to amend?

**Reply by DauntlessEagle** [reply]
![IMG_3918](https://github.com/svenerbe/ecoflow_dynamic_power_adjustment/assets/145854497/ed30f7be-8f4a-4e0e-84d2-683db5fd43f8)

**Reply by bragg2012** [reply]
what does your automation config look like?
